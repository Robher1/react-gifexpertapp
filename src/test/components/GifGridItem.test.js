import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', ()=>{
    
    const title = 'Soy un test';
    const url = 'https://localhost/soyuntest.gif';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );
    
    test('debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe tener un parrafo', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe( title );
        
    });

    test('debe de obtener la imagen igual al url y alt de los props', async () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
            
    });

    test('debe de tener animate__fadeIn', async () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );
        
    });
    
});