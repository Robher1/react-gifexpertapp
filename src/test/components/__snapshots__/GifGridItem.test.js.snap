// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy un test"
    src="https://localhost/soyuntest.gif"
    title="Soy un test"
  />
  <p>
    Soy un test
  </p>
</div>
`;
